lib/ffi.dart

```diff
@@ -40,6 +40,29 @@ Future<User> ffiUserData() async {
   return User.create()..mergeFromProto3Json(jsonDecode(res));
 }
 
+// checkAPIError throws a PlatformException if the API response contains an error
+void checkAPIError(result, errorMessage) {
+  if (result.error != "") {
+    throw PlatformException(code: result.error, message: errorMessage);
+  }
+}
+
+Future<String> ffiApproveDevice(code) async {
+  final json = await _bindings.approveDevice(code).cast<Utf8>().toDartString();
+  final result = APIResponse.create()..mergeFromProto3Json(jsonDecode(json));
+  checkAPIError(result, 'wrong_device_linking_code'.i18n);
+  // refresh user data after successfully linking device
+  await ffiUserData();
+  return json;
+}
+
+Future<void> ffiRemoveDevice(deviceId) async {
+  final json = await _bindings.removeDevice(deviceId).cast<Utf8>().toDartString();
+  final result = LinkResponse.create()..mergeFromProto3Json(jsonDecode(json));
+  checkAPIError(result, 'cannot_remove_device'.i18n);
```

Comment:
@atavism After removing, I think we need to update device list as well?

Reply:
Done, thanks!